src/lib/TypeExtensions/ApiManifestDocumentExtensions.cs

```diff
@@ -0,0 +1,69 @@
+        public static async Task<OpenAIPluginManifest> ToOpenAIPluginManifestAsync(this ApiManifestDocument apiManifestDocument, string logoUrl, string legalInfoUrl, string? apiDependencyName = default, string openApiFilePath = "./openapi.json", CancellationToken cancellationToken = default)
+        {
+            if (!TryGetApiDependency(apiManifestDocument.ApiDependencies, apiDependencyName, out ApiDependency? apiDependency))
+            {
+                throw new ApiManifestException(string.Format(ErrorMessage.ApiDependencyNotFound, nameof(OpenAIPluginManifest)));
+            }
+            else if (string.IsNullOrWhiteSpace(apiDependency?.ApiDescriptionUrl))
+            {
+                throw new ApiManifestException(string.Format(ErrorMessage.ApiDescriptionUrlNotFound, nameof(OpenAIPluginManifest)));
+            }
+            else
+            {
+                var result = await ParsingHelpers.ParseOpenApiAsync(apiDependency.ApiDescriptionUrl, false, cancellationToken);
+                return apiManifestDocument.ToOpenAIPluginManifest(openApiDocument: result.OpenApiDocument, logoUrl: logoUrl, legalInfoUrl: legalInfoUrl, openApiFilePath: openApiFilePath);
+            }
+        }
+
+        internal static OpenAIPluginManifest ToOpenAIPluginManifest(this ApiManifestDocument apiManifestDocument, OpenApiDocument openApiDocument, string logoUrl, string legalInfoUrl, string openApiFilePath)
+        {
+            // Validate the ApiManifestDocument before generating the OpenAI manifest.
+            apiManifestDocument.Validate();
+            string contactEmail = string.IsNullOrWhiteSpace(apiManifestDocument.Publisher?.ContactEmail) ? string.Empty : apiManifestDocument.Publisher.ContactEmail;
+
+            var openApiManifest = OpenApiPluginFactory.CreateOpenAIPluginManifest(
+                schemaVersion: openApiDocument.Info.Version,
+                nameForHuman: openApiDocument.Info.Title,
+                nameForModel: openApiDocument.Info.Title,
+                logoUrl: logoUrl,
+                contactEmail: contactEmail,
+                legalInfoUrl: legalInfoUrl);
+
+            openApiManifest.Api = new Api("openapi", openApiFilePath);
+            openApiManifest.Auth = new ManifestNoAuth();
+            openApiManifest.DescriptionForHuman = openApiDocument.Info.Description ?? $"Description for {openApiManifest.NameForHuman}.";
+            openApiManifest.DescriptionForModel = openApiManifest.DescriptionForHuman;
+
+            return openApiManifest;
+        }
+
+        private static bool TryGetApiDependency(ApiDependencies apiDependencies, string? apiDependencyName, out ApiDependency? apiDependency)
+        {
+            if (apiDependencyName == default)
+                apiDependency = apiDependencies.FirstOrDefault().Value;
+            else
+                _ = apiDependencies.TryGetValue(apiDependencyName, out apiDependency);
+            return apiDependency != null;
```

Comment:
In this case, the caller will throw an exception with ApiDependencyNotFound since the method yielded a null apiDependency (the method returned false). The method is intended to only return true when it can retrieve an apiDependency that's not null. I'll add this as a code comment to make it clear.